layerindex/views.py

```diff
@@ -944,13 +1464,493 @@ def get_context_data(self, **kwargs):
+        context['existing_cover_recipe'] = recipe.get_cover_recipe()
+        comparison_form = ClassicRecipeForm(prefix='selectrecipedialog', instance=recipe)
+        comparison_form.fields['cover_pn'].widget = forms.HiddenInput()
+        comparison_form.fields['cover_layerbranch'].widget = forms.HiddenInput()
+        context['comparison_form'] = comparison_form
+        context['can_edit'] = False
+        return context
+
+    def post(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            raise PermissionDenied
+
+        recipe = get_object_or_404(ClassicRecipe, pk=self.kwargs['selectfor'])
+        form = ClassicRecipeForm(request.POST, prefix='selectrecipedialog', instance=recipe)
+
+        if form.is_valid():
+            form.save()
+            messages.success(request, 'Changes to comparison recipe %s saved successfully.' % recipe.pn)
+            return HttpResponseRedirect(reverse('comparison_recipe', args=(recipe.id,)))
+        else:
+            # FIXME this is ugly because HTML gets escaped
+            messages.error(request, 'Failed to save changes: %s' % form.errors)
+
+        return self.get(request, *args, **kwargs)
+
+
+class BranchCompareView(FormView):
+    form_class = BranchComparisonForm
+
+    def get_recipes(self, from_branch, to_branch, layer_ids):
+        from distutils.version import LooseVersion
+        class BranchComparisonResult:
+            def __init__(self, pn, short_desc):
+                self.pn = pn
+                self.short_desc = short_desc
+                self.from_versions = []
+                self.to_versions = []
+                self.id = None
+            def pv_changed(self):
+                from_pvs = sorted([x.pv for x in self.from_versions])
+                to_pvs = sorted([x.pv for x in self.to_versions])
+                return (from_pvs != to_pvs)
+        class BranchComparisonVersionResult:
+            def __init__(self, id, pv, srcrev):
+                self.id = id
+                self.pv = pv
+                self.srcrev = srcrev
+            def version_expr(self):
+                return (self.pv, self.srcrev)
+
+        def map_name(recipe):
+            pn = recipe.pn
+            if pn.startswith('gcc-source-'):
+                pn = pn.replace('-%s' % recipe.pv, '')
+            elif pn.endswith(('-i586', '-i686')):
+                pn = pn[:-5]
+            elif pn.endswith('-x86_64-oesdk-linux'):
+                pn = pn[:-19]
+            return pn
+
+        from_recipes = Recipe.objects.filter(layerbranch__branch=from_branch)
+        to_recipes = Recipe.objects.filter(layerbranch__branch=to_branch)
+        if layer_ids:
+            from_recipes = from_recipes.filter(layerbranch__layer__in=layer_ids)
+            to_recipes = to_recipes.filter(layerbranch__layer__in=layer_ids)
+        recipes = {}
+        for recipe in from_recipes:
+            pn = map_name(recipe)
+            res = recipes.get(pn, None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)

Similar issue at line number 1784.